### PR TITLE
feat: selectable items and pause on hover

### PR DESCRIPTION
**Pull request by mdjfs**
`make selectable items, and pause when any item is hover`

Thanks for the guide, here is the PR, if we need any changes feel free to edit or suggest that, thank you !

### CODE REVIEW

File: src/components/Carousel/Carousel.tsx
Comment (Byongho96) [inline src/components/Carousel/Carousel.tsx:53]: I initially suggested adding the `onClickItem` function, but later realized it is unnecessary. Users can directly set up the `onClick` event handler on each carousel item when passing the items as props

File: src/components/Carousel/Carousel.tsx
Comment (Byongho96) [inline src/components/Carousel/Carousel.tsx:118]: For the same reason, `onClickItem` can also be removed here.

File: src/components/Carousel/Carousel.tsx
Comment (Byongho96) [inline src/components/Carousel/Carousel.tsx:168]: In my opinion, it would be beneficial if 'selectable' and 'onClickSelectedItem' are independent.
- 'Selectable' could handle whether the curIndex can be changed through a click
- 'onClickSelectedItem' is the event when the currently selected (frontmost) carousel item is clicked
```javascript
(index: number) => {
    if (selectable) setCurIndex(index)
    if (onClickSelectedItem && curIndex === index) onClickFocusedItem(index)
  }
```

File: src/components/Carousel/Carousel.tsx
Comment (Byongho96) [inline src/components/Carousel/Carousel.tsx:179]: This is my opinion, but how about attaching the pauseHandlers(`startPause`, `releasePause`) to the carousel container instead of each individual carousel item?

```javascript
<div
   className={`react-responsive-3d-carousel__carousel__list ${spread}`}
   ref={containerRef}
   onPointerEnter={startPause}
   onPointerLeave={releasePause}
 >
```

```javascript
  const startPause = useCallback(() => {
    if (pauseOnHover) pause.current = true
  }, [pauseOnHover])

  const releasePause = useCallback(() => {
    if (pauseOnHover) pause.current = false
  }, [pauseOnHover])
```

File: src/components/Carousel/Carousel.tsx
Comment (Byongho96) [inline src/components/Carousel/Carousel.tsx:249]: The `if (!pause.current) {}` condition can be removed because this useEffect function is triggered `by curIndex`. And the functions (SlidePrev and SlideNext) that change the `curIndex` are already prevented when `pause.current === true`.

**Review by Byongho96 — COMMENTED** [review]
Again, it's a great feature. I would like to use the added features with their default settings enabled. I've left a few comments, so please check them.

File: src/components/Carousel/Carousel.tsx
Comment (mdjfs) [reply src/components/Carousel/Carousel.tsx]: totally, changed

File: src/components/Carousel/Carousel.tsx
Comment (mdjfs) [reply src/components/Carousel/Carousel.tsx]: idem

File: src/components/Carousel/Carousel.tsx
Comment (mdjfs) [reply src/components/Carousel/Carousel.tsx:166]: I agree to make independent the event and the selectable property. Done

File: src/components/Carousel/Carousel.tsx
Comment (mdjfs) [reply src/components/Carousel/Carousel.tsx]: My question here is if you want to use the arrows to change the slide and don't work, but I tried it and it worked

File: src/components/Carousel/Carousel.tsx
Comment (mdjfs) [reply src/components/Carousel/Carousel.tsx]: Done !

**Comment by mdjfs** [comment]
Hello! Happy new year in Korea! 

Thanks for the review! Here is the changes before the new year 2024 (in my case)

image

**Review by Byongho96 — APPROVED** [review]
Thank you for the changes from 2023, lol!
I'll update the package soon. Happy New Year to you too!

**Comment by Byongho96** [comment]
Okay so there's a small issue that you doesn't appear in the contributor tab after the merge
image

I checked and found that the commit is made with another account.
image

After googling, it seems that your local computer's GitHub email settings may differ from the actual email associated with your GitHub account. [How to set your commit email address](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#about-commit-email-addresses)

**Comment by Byongho96** [comment]
Personally, I'd love to see you on the contributors tab. If you wish for that as well, **I'll revert the recent merge**, and **you can create the same pull request again**. Please make sure the commits go through correctly under your account.

Just leave your decision as a comment. If you agree, I'll revert the commit right away.

**Comment by mdjfs** [comment]
Hi! Happy new year! Yeah i'm sorry! If you can revert the last merge i will send a new PR with my account. Sorry for that

**Comment by Byongho96** [comment]
No problem:) I just reverted the commits. You may have to pull again to avoid conflicts

**Pull request merged**
